app/controllers/api/v1/users_controller.rb

```diff
@@ -27,6 +27,19 @@ def show
       end
     end
 
+    def search
+      if params[:q].present?
+        users = User.where("name ~* ?", params[:q])
+      else
+        users = User.none
+      end
+      users = users.by_email
```

Comment:
But surely the whole point is that all the records which are returned have the same name... :stuck_out_tongue: